cmsis/CMSIS_5/CMSIS/RTOS2/RTX/Source/TOOLCHAIN_GCC/TARGET_M0/irq_cm0.S

```diff
@@ -102,7 +102,7 @@ SVC_ContextRestore:
         SUBS     R0,R0,#32              // Adjust address
         LDMIA    R0!,{R4-R7}            // Restore R4..R7
 
-        MOVS     R0,#~0xFFFFFFFD
+        MOVS     R0,#2                  // Binary complement of 0xFFFFFFFD
```

Comment:
Since this file is for M0, does it mean we've decided to use GNU-style assembly for all cores and not just M55? If this is the case, could we remove all the legacy armasm assembly files and rename/remove the intermediate path `TOOLCHAIN_GCC`?

Reply:
The legacy tools only use `armasm` for compiling assembly, so the files are needed to avoid breaking those tools.

Reply:
> The legacy tools only use `armasm` for compiling assembly, so the files are needed to avoid breaking those tools.

Fair enough. We should've switched to `armclang` a long time ago with the legacy tools, but it's frozen so there's not much we could do...